Merging front end with back end for movie card and modal

FadyAPhilips opened the pull request:
Made other styling changes to many components

FadyAPhilips committed:
made frontend changes with the movie card and movie modal

FadyAPhilips committed:
Merge branch 'master' into mergingFrontEndWithBackEndForMovieCardAndModal

FadyAPhilips committed:
Integreated backend with the movie poster and modal also add search functionality

also made many small styling changes to many components

markarmanus commented on src/Routes/Home.js:
```diff
@@ -1,12 +1,79 @@
+        trailerID: movieDetails.data.videos.results[0].key,
+      };
+      this.setState({ showModal: true, modalData: movieData });
+    }, 1000);
```
play around with this.

markarmanus commented on src/Routes/Home.js:
```diff
@@ -1,12 +1,79 @@
+    }}
+    spin
+  />
+);
```
move to a component that takes in size as a prop.

markarmanus commented on src/Routes/Home.js:
```diff
@@ -1,12 +1,79 @@
+    this.setState({ showModal: false });
+  };
+
+  getFirstFiveNames(obj) {
```
move to a movie transformer file.

markarmanus commented on src/Routes/Home.js:
```diff
@@ -17,28 +84,88 @@ class Home extends React.Component {
+      });
+  }
+  renderMovieCards() {
+    const movies = [];
```
move this to a .map

markarmanus commented on src/Routes/Home.js:
```diff
@@ -17,28 +84,88 @@ class Home extends React.Component {
+            showSearchBar={true}
+          ></NavBar>
+          <BlurDiv
+            style={{ height: "100%" }}
```
figure out the height 100%

markarmanus commented on src/Routes/Home.js:
```diff
@@ -17,28 +84,88 @@ class Home extends React.Component {
+              title={"Movies"}
+            ></ExpandingDivider>
+            <OnImagesLoaded
+              key={this.state.index}
```
ijdex

markarmanus commented on src/Routes/Home.js:
```diff
@@ -17,28 +84,88 @@ class Home extends React.Component {
+              }
+              timeout={300}
+            >
+              <FadeIn key={this.state.index}>
```
index

FadyAPhilips committed:
cleaned up code and fixed errors

created a Helper file which has a class containing methods to be used everywhere
The helper class currently has a helper method which takes in the information for a movie from the back end and transforms it into information which could be displayed in the movie modal.
Created an image to be displayed when no poster is found. This fixed the error that comes up when trying to access a movie that has no poster.
fixed error that will come up when trying to access a movie without a trailer. replaced by video id 'NA'

Changed the cast to take in a number n of how many cast members would be displayed. fixed error which would come up when not enough cast members are retrieved from the backend.

moved the loading spinner into its own component.

Made general cleanups to the code

markarmanus commented on src/Helper.js:
```diff
@@ -0,0 +1,40 @@
+        .join(", "),
+      releaseDate: movieDetails.release_date,
+      cast:
+        movieDetails?.credits?.cast?.length > 0
```
move this to the getFirstNCast Fucntion

markarmanus commented on src/Helper.js:
```diff
@@ -0,0 +1,40 @@
+          : "N/A",
+      info: movieDetails.overview,
+      trailerID: movieDetails.videos.results[0]
+        ? movieDetails.videos.results[0].key
```
movieDetails.videos.results[0]?.key || "NA"

markarmanus commented on src/Routes/Home.js:
```diff
@@ -41,37 +27,10 @@ class Home extends React.Component {
-      };
-      this.setState({ showModal: true, modalData: movieData });
-    }, 1000);
+    const movieData = Helper.movieTransformer(movieDetails.data);
```
make the name clearer.

markarmanus commented on src/UI/MovieCard.js:
```diff
@@ -131,7 +133,10 @@ const MovieCard = (props) => {
       <CardContainer onMouseEnter={toggleHover} onMouseLeave={toggleHover}>
-        <PosterStyle draggable={false} src={props.posterSrc} />
+        <PosterStyle
+          draggable={false}
```
move main check here
